src/index.ts

```diff
@@ -220,27 +240,35 @@ export default class SwupScrollPlugin extends Plugin {
 		return true;
 	}
 
+	/**
+	 * Prepare scrolling before visit:start hook
+	 */
+	onBeforeVisitStart: Handler<'visit:start'> = (visit) => {
+		visit.scroll.scrolledToContent = false;
+		visit.scroll.animate = this.shouldAnimate('betweenPages');
+	};
+
 	/**
 	 * Check whether to scroll in `visit:start` hook
 	 */
 	onVisitStart: Handler<'visit:start'> = (visit) => {
 		this.maybeResetScrollPositions(visit);
 		this.cacheScrollPositions(visit.from.url);
 
-		const scrollTarget = visit.scroll.target || visit.to.hash;
+		const scrollTarget = visit.scroll.target ?? visit.to.hash;
 
-		visit.scroll.scrolledToContent = false;
-
-		if (this.options.doScrollingRightAway && !scrollTarget) {
-			visit.scroll.scrolledToContent = true;
+		// Conditions for scrolling before content replace:
+		// - scroll is animated (otherwise the effect is useless)
+		// - no scroll target is defined (needs to wait until new content is there)
+		if (visit.scroll.animate && this.options.doScrollingRightAway && !scrollTarget) {
```

Comment:
Added condition: only allow scrolling before animation if the scroll is animated as well, otherwise the effect is useless.